app/controllers/api/v1/authentication_controller.rb

```diff
@@ -29,6 +27,44 @@ def signup
     end
   end
 
+  # POST api/v1/oauth/login
+  def oauth_login
+    @user = User.find_by!(email: params["email"])
+    token = JsonWebToken.encode(
+      user_id: @user.id, username: @user.name, email: @user.email
+    )
+    render json: { token: token }, status: :accepted
```

Comment:
Wait, you can pass arbitrary email addresses and this endpoint will return a valid token for that user !!. Big security flow

Reply:
@tachyons is that so? what we are doing here is just registering or finding the user with that email/info in the DB.. & imo devise handles email validation on its own, so it won't allow any email to generate a valid token..
Correct me if i am wrong..

Reply:
@tachyons

Reply:
Where are you validating that ?

Reply:
@tachyons pushed the appropriate changes discussed, can you please once check.. if given a green signal, will add tests for the same..